### ISSUE TITLE
Problems with non-combat formations in StratCon

### ISSUE THREAD

**Issue by SuperStucco** [MegaMek/mekhq#2749]
In 49.2, formations flagged as non-combat are being selected as primary forces for scenarios.  In a few cases this is appropriate e.g. as part of a convoy in a Convoy Escort scenario, or in a Base Defense scenario.  However they are also being assigned as primary forces for other scenarios, as in the attached campaign.

[Jansenn's Horde30420214.cpnx.gz](https://github.com/MegaMek/mekhq/files/6831692/Jansenn.s.Horde30420214.cpnx.gz)
[Customs.zip](https://github.com/MegaMek/mekhq/files/6831694/Customs.zip)

There are other problems as a result of this.  Because the formation is flagged as non-combat they are not always assigned to the scenario i.e. the scenario is generated and no forces are assigned; in this specific example, because the primary formation is non-combat the map size is 0x0 and no defending forces are generated.

Sometimes a scenario has no primary force assigned but the victory conditions still list the non-combat formation.  While this formation could be left out and a combat formation assigned, in the scenario resolution after playing it out in MegaMek the units do not have check boxes in the victory conditions window leaving it as an automatic fail.  If the AtB Campaign State "Manage Scenario" dialog is used to assign the non-combat formation to the scenario, because it is flagged as non-combat it is not deployed to MegaMek and the units are still not in the scenario resolution window.  The only way for this to function is to read the victory conditions thoroughly, find the listed formations, ensure they are flagged as combat, and manually deploy them from the TO&E tab.

Short version: non-combat formations should be handled better in StratCon.  There needs to be a point to the combat/non-combat distinction.  They should not be assigned as primary forces.  They could be assigned as secondary forces in scenarios such as Convoy Escort or Base Defense.  When scenario victory conditions include preserving part of a formation, those formations (combat or non-combat) should be automatically added to the scenario and also sent to MegaMek.

**Comment by UlyssesSockdrawer** [comment]
Should be fixed by https://github.com/MegaMek/mekhq/pull/5010 - can be closed once merged.

**Comment by IllianiCBT** [comment]
Closing as resolved
